src/workspace-element.js

```diff
@@ -222,21 +222,101 @@ class WorkspaceElement extends HTMLElement {
     this.model.getActivePane().activate()
   }
 
-  focusPaneViewAbove () { this.paneContainer.focusPaneViewAbove() }
+  focusPaneViewAbove () { this.focusPaneViewInDirection('above') }
 
-  focusPaneViewBelow () { this.paneContainer.focusPaneViewBelow() }
+  focusPaneViewBelow () { this.focusPaneViewInDirection('below') }
 
-  focusPaneViewOnLeft () { this.paneContainer.focusPaneViewOnLeft() }
+  focusPaneViewOnLeft () { this.focusPaneViewInDirection('left') }
 
-  focusPaneViewOnRight () { this.paneContainer.focusPaneViewOnRight() }
+  focusPaneViewOnRight () { this.focusPaneViewInDirection('right') }
 
-  moveActiveItemToPaneAbove (params) { this.paneContainer.moveActiveItemToPaneAbove(params) }
+  focusPaneViewInDirection (direction, pane) {
+    const activePane = this.model.getActivePane()
+    const paneToFocus = this.nearestVisiblePaneInDirection(direction, activePane)
+    paneToFocus && paneToFocus.focus()
+  }
+
+  moveActiveItemToPaneAbove (params) {
+    this.moveActiveItemToNearestPaneInDirection('above', params)
+  }
 
-  moveActiveItemToPaneBelow (params) { this.paneContainer.moveActiveItemToPaneBelow(params) }
+  moveActiveItemToPaneBelow (params) {
+    this.moveActiveItemToNearestPaneInDirection('below', params)
+  }
 
-  moveActiveItemToPaneOnLeft (params) { this.paneContainer.moveActiveItemToPaneOnLeft(params) }
+  moveActiveItemToPaneOnLeft (params) {
+    this.moveActiveItemToNearestPaneInDirection('left', params)
+  }
+
+  moveActiveItemToPaneOnRight (params) {
+    this.moveActiveItemToNearestPaneInDirection('right', params)
+  }
+
+  moveActiveItemToNearestPaneInDirection (direction, params) {
+    const activePane = this.model.getActivePane()
+    const nearestPaneView = this.nearestVisiblePaneInDirection(direction, activePane)
+    if (nearestPaneView == null) { return }
+    if (params && params.keepOriginal) {
+      activePane.getContainer().copyActiveItemToPane(nearestPaneView.getModel())
+    } else {
+      activePane.getContainer().moveActiveItemToPane(nearestPaneView.getModel())
+    }
+    nearestPaneView.focus()
+  }
+
+  nearestVisiblePaneInDirection (direction, pane) {
+    const distance = function (pointA, pointB) {
+      const x = pointB.x - pointA.x
+      const y = pointB.y - pointA.y
+      return Math.sqrt(Math.pow(x, 2) + Math.pow(y, 2))
+    }
 
-  moveActiveItemToPaneOnRight (params) { this.paneContainer.moveActiveItemToPaneOnRight(params) }
+    const paneView = pane.getElement()
+    const box = this.boundingBoxForPaneView(paneView)
+
+    const visiblePaneContainers = atom.workspace.getPaneContainers()
+      .filter(container => {
+        const isCenter = container === atom.workspace.getCenter()
+        return isCenter || container.isVisible()
+      })
+
+    const visiblePanes = _.flatten(visiblePaneContainers.map(container => container.getPanes()))
```

Comment:
I like the idea of a getVisiblePanes function and to me Workspace is the place to put it given that other pane functions are there such as getPanes - one option would be to overload the latter with a "onlyVisible" boolean parameter if we want to keep the API surface down - I'm not convinced one way or the other there.